homeassistant/components/nice_go/coordinator.py

```diff
@@ -220,4 +227,39 @@ async def on_data(self, data: dict[str, Any]) -> None:
     async def on_connected(self) -> None:
         """Handle the websocket connection."""
         _LOGGER.debug("Connected to the websocket")
+        self.connected = True
+
         await self.api.subscribe(self.organization_id)
+
+        if not self.last_update_success:
+            self.async_set_updated_data(self.data)
+
+    async def on_connection_lost(self, data: dict[str, Exception]) -> None:
+        """Handle the websocket connection loss. Don't need to do much since the library will automatically reconnect."""
+        _LOGGER.debug("Connection lost to the websocket")
+        self.connected = False
+
+        # Give some time for reconnection
+        await asyncio.sleep(5)
+        if self.connected:
+            _LOGGER.debug("Reconnected, not setting error")
+            return
+
+        # There's likely a problem with the connection, and not the server being flaky
+        self.async_set_update_error(data["exception"])
+
+    def unsubscribe(self) -> None:
+        """Unsubscribe from the websocket."""
+        if (
+            self._unsub_connected is None
+            or self._unsub_data is None
+            or self._unsub_connection_lost is None
+        ):
+            return
```

Comment:
Look like you are missing a test case for this line

Reply:
Couldn't figure out how to test this without touching `entry.runtime_data`

Reply:
Not sure when it is called, but based on the on the entry unload test you could mock the setup to fail, and then call unload entry unload?

Reply:
Right, then do we need the check in that case. If it is only to assure that they are not `None` for typing you could use `assert` inside of a `if TYPE_CHECKING` block instead.